tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -331,3 +336,144 @@ def _transform_access_policies(access_policies):
                 "permissions": i['permissions']} for i in access_policies]
 
         return {"accessPolicies": policies}
+
+
+@KeyVault.action_registry.register('update')
+class KeyVaultUpdateAction(AzureBaseAction):
+    """
+    Update a keyvault
+
+    To keep the tenant id the same, set it to "keep"
+    To keep the sku settings the same, set it to {"current": True}
+
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: keyvault-vault-disable-public-access
+              resource: azure.keyvault
+              filters:
+                - name: test
+              actions:
+                - type: update
+                  configuration:
+                    tenant_id: keep
+                    sku:
+                      current: True
+                    public_network_access: disabled
+    """
+
+    @staticmethod
+    def generate_schema():
+        from azure.identity import DefaultAzureCredential
+        from azure.mgmt.keyvault import KeyVaultManagementClient
+
+        sub_id = "00000000-0000-0000-0000-000000000000"
+        client = KeyVaultManagementClient(
+            credential=DefaultAzureCredential(),
+            subscription_id=sub_id
+        )
+        vault_properties = client.models().VaultProperties
+        model_signature = inspect.signature(vault_properties)
+
+        schema_dict = {}
+        required = ["tenant_id", "sku"]
+
+        for name, v in model_signature.parameters.items():
```

Comment:
this method feels like its doing quite a lot of meta including client construction, why not just do something simple like pass through on a kw dictionary field. Ala type: object

Reply:
simple pass through here hinders discoverability on settings you can use here, having gone through the azure docs for these PRs it's pretty opaque/hard to use...

There might be an easier way to do this, when looking at the source code in the azure sdk there are some private variables on the model classes I could pull out and reconstruct the jsonschema that way, but I'm hesitant to reach into private class variables

Reply:
though we can also provide a list of valid values as well, .e.g `{"type": "object", "properties": {"name": {"type": "string", "enum": [...]}}`

Reply:
ack, it would still be good to reference the azure docs on the action docs